broker.go

```diff
@@ -390,24 +390,35 @@ func (b *Broker) coordinatorConnection(consumerGroup string) (conn *connection,
 		}
 
 		if conn != nil {
-			defer func(lconn *connection) { go b.conns.Idle(lconn) }(conn)
-			resp, err := conn.ConsumerMetadata(&proto.ConsumerMetadataReq{
-				ClientID:      b.conf.ClientID,
-				ConsumerGroup: consumerGroup,
-			})
+			// Run this in a closure so we can ensure we release the connection
```

Comment:
Probably easier to read as a real function considering that this closure interacts with `conn` that then gets overwritten on line 423 but I don't feel strongly.